typescript/sdk/src/gas/token-prices.ts

```diff
@@ -8,8 +8,12 @@ import { ChainMetadata } from '../metadata/chainMetadataTypes';
 import { ChainMap, ChainName } from '../types';
 
 export interface TokenPriceGetter {
-  getTokenPrice(chain: ChainName): Promise<number>;
-  getTokenExchangeRate(base: ChainName, quote: ChainName): Promise<number>;
+  getTokenPrice(chain: ChainName, cacheThemAll: ChainName[]): Promise<number>;
+  getTokenExchangeRate(
+    base: ChainName,
+    quote: ChainName,
+    cacheThemAll: ChainName[],
```

Comment:
Yes, but the script was failing before this, now sure what you want me to do here otherwise. Minor changeset?

PLS SUGGEST ALTERNATIVES EVERYTHING YOU COMMENT ABOUT VARIABLE NAMES.

Reply:
is it `chainsToCache` or does it mean something else?

Reply:
can you make this optional? I would have no idea from the name how to call `getTokenPrice` after this change